Livecheckables/go@1.14.rb

```diff
@@ -0,0 +1,6 @@
+class GoAT114
+  livecheck do
+    url "https://golang.org/dl/"
+    regex(/href=.*?gov?(1\.14(?:\.\d+)+)\.src/i)
```

Comment:
```suggestion
    regex(/href=.*?gov?(1\.14(?:\.\d+)*)\.src/i)
```

The first Go release for a given minor release only includes the major and minor (no patch). Since this regex specifies the major and minor version, we should make the patch optional (i.e., `(?:\.\d+)*` instead of `(?:\.\d+)+`).

Reply:
> I've also made some other changes to bring this more in line with our current standards.  I'll create a PR to update the `go` livecheckable in the same fashion.

👍